Dockerfile

```diff
@@ -0,0 +1,34 @@
+FROM alpine
+
+RUN apk --no-cache add gnupg \
+    build-base \
+    gettext-dev \
+    libgcrypt-dev \
+    zlib-dev \
+    gnutls-dev \
+    curl-dev \
+    cmake \
+    aspell-dev \
+    ncurses-dev
+
+RUN wget https://weechat.org/files/src/weechat-2.8.tar.xz \
+    && wget https://weechat.org/files/src/weechat-2.8.tar.xz.asc \
+    && gpg --keyserver hkps://keys.openpgp.org --recv-keys A9AB5AB778FA5C3522FD0378F82F4B16DEC408F8 \
+    # Check that the fingerprint is correct by checking the topic on #weechat channel (chat.freenode.net) 
+    # or by asking a developer.
+    && gpg --verify weechat-2.8.tar.xz.asc weechat-2.8.tar.xz \
+    && tar xvf weechat-2.8.tar.xz \
+    && mkdir -p weechat-2.8/build
+
+WORKDIR /weechat-2.8/build
+RUN cmake .. -DENABLE_PHP=OFF \
+    -DENABLE_JAVASCRIPT=OFF \
+    -DENABLE_PYTHON=OFF \
+    -DENABLE_RUBY=OFF \
+    -DENABLE_LUA=OFF \
+    -DENABLE_PERL=OFF \
+    -DENABLE_TCL=OFF \
+    -DENABLE_GUILE=OFF
```

Comment:
>  Some script dependencies are difficult to add to Alpine such as V8

Javascript plugin (based on V8) is problematic in general (issue #360) and also disabled in our Debian builds, if I remember correctly. Other plugins should be fine though and likely should be enabled. Especially Python and Perl plugins, which are required for using popular scripts. Users of a Docker container, who want easy and reproducible WeeChat setup, would likely want to also use scripts as easily.

Reply:
As @sim642 said, I think all plugins except Javascript must be enabled (not only Python and Perl).
If debian base is better for this, then it can be used (I'm more familiar with Debian than Alpine, but I guess Alpine is used here for image size reason only).

Reply:
Additional note about Javascript: I'm currently working on disabling this plugin by default, as it compiles only with an old libv8 version, I don't have enough time myself to fix/rewrite the plugin, and got no help past years on this subject (see #360 for more information).

Reply:
Done: JavaScript plugin is now disabled by default: 340d6646a6371e2f224d392c9b1b44c8a20b5074.

Reply:
I'm probably more familiar with Debian myself also, but Alpine does have some nice security features and smaller image size.  Now that JS is disabled, I could update this to just use the default `cmake` with either alpine or debian depending on which seems more feasible to support.

Reply:
Added all script deps to the build process.  I had some trouble getting Cmake to find `php-dev` so I switched to `debian:stable-slim` which has a fairly small image size as well.